packages/main/src/ColorPalette.js

```diff
@@ -209,16 +254,131 @@ class ColorPalette extends UI5Element {
+
+		if (isUp(event)) {
+			event.stopPropagation();
+
+			this.focusColorPaletteElement(this.displayedColors[colorPaletteFocusIndex]);
+		} else if (isDown(event)) {
+			event.stopPropagation();
+
+			if (hasRecentColors) {
+				this.focusRecentColorsElement(this.colorPaletteNavigationElements[index + 1]);
+			} else if (this.showDefaultColor) {
+				this.colorPaletteNavigationElements[0].focus();
+			} else {
+				this.focusColorPaletteElement(this.displayedColors[0]);
+			}
+		}
+	}
+
+	_onColorContainerKeyDown(event) {
+		const hasRecentColors = this.showRecentColors && this.recentColors[0];
+		const lastElementInNavigation = this.colorPaletteNavigationElements[this.colorPaletteNavigationElements.length - 1];
+		if (isTabNext(event) && this.popupMode) {
+			event.preventDefault();
 			this.selectColor(event.target);
 		}
+
+		if (isUp(event) && event.target === this.displayedColors[0] && this.colorPaletteNavigationElements.length > 1) {
+			event.stopPropagation();
+			if (this.showDefaultColor) {
+				this.colorPaletteNavigationElements[0].focus();
+			} else if (!this.showDefaultColor && hasRecentColors) {
+				this.focusRecentColorsElement(lastElementInNavigation);
+			} else if (!this.showDefaultColor && this.showMoreColors) {
+				lastElementInNavigation.focus();
+			}
+		} else if (isDown(event) && event.target === this.displayedColors[this.displayedColors.length - 1] && this.colorPaletteNavigationElements.length > 1) {
+			event.stopPropagation();
+			const isRecentColorsNextElement = (this.showDefaultColor && !this.showMoreColors && hasRecentColors) || (!this.showDefaultColor && !this.showMoreColors && hasRecentColors);
+
+			if (this.showDefaultColor && this.showMoreColors) {
+				this.colorPaletteNavigationElements[2].focus();
+			} else if (this.showDefaultColor && !this.showMoreColors && (!this.showRecentColors || !this.recentColors[0])) {
+				this.colorPaletteNavigationElements[0].focus();
+			} else if (isRecentColorsNextElement) {
+				this.focusRecentColorsElement(lastElementInNavigation);
+			} else if (!this.showDefaultColor && this.showMoreColors) {
+				this.colorPaletteNavigationElements[1].focus();
+			}
+		}
+	}
+
+	_onRecentColorsContainerKeyDown(event) {
+		if (isUp(event)) {
+			if (this.showMoreColors) {
+				this.colorPaletteNavigationElements[1 + this.showDefaultColor].focus();
+			} else if (!this.showMoreColors && this.colorPaletteNavigationElements.length > 1) {
+				const rowSize = 5;
+				const colorPaletteFocusIndex = (this.displayedColors.length % rowSize) * rowSize;
+				event.stopPropagation();
+
+				this.focusColorPaletteElement(this.displayedColors[colorPaletteFocusIndex]);
+			}
+		} else if (isDown(event)) {
+			if (this.showDefaultColor) {
+				this.colorPaletteNavigationElements[0].focus();
+			} else {
+				event.stopPropagation();
+				this.focusColorPaletteElement(this.displayedColors[0]);
+			}
+		}
+	}
+
+	focusRecentColorsElement(element) {
```

Comment:
Can you please declare one function instead of this and ```focusRecentColorsElement``` and use it respectively.